services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3InternalSdkHttpExecutionAttribute.java

```diff
@@ -41,6 +43,13 @@ public final class S3InternalSdkHttpExecutionAttribute<T> extends SdkHttpExecuti
     public static final S3InternalSdkHttpExecutionAttribute<Region> SIGNING_REGION =
         new S3InternalSdkHttpExecutionAttribute<>(Region.class);
 
+    public static final S3InternalSdkHttpExecutionAttribute<Path> SOURCE_REQ_PATH =
+        new S3InternalSdkHttpExecutionAttribute<>(Path.class);
```

Comment:
Naming nitpicks, since this actually seems like a protected API.
* `SOURCE` - Do we ever expect to use this for something other than a PutObject - like a GetObject, where it's actually the `DESTINATION`?
* `REQ` - We should almost always use the full word for something instead of shortening it, so `REQUEST`. It looks like a lot of the existing properties are request properties, though, so I don't think REQUEST is necessary.

If it's always the source, what about `PUT_OBJECT_PATH`? `UPLOAD_PATH`?
If it's potentially also destination, what about `UPLOAD_DOWNLOAD_PATH`?

Reply:
Thanks Matt..
Named it as OBJECT_FILE_PATH so that it can be used for any file path for get/put